### PR TITLE
Add finally task to set commit status

### PR DESCRIPTION
**Pull request by chetan-rns**
**What type of PR is this?**

> /kind enhancement

**What does this PR do / why we need it**:

This PR removes references to the commit status tracker controller and adds finally tasks to set commit status. For the task to set status, the `git-host-access-token` flag will be mandatory.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

1. Bootstrap the GitOps repository on GitHub/GitLab
2. Add webhooks for GitOps and taxi repository
3. Raise PRs and observe commit status being set
        1. Initial status will be set to "pending"
        2. Final status will be set after the execution of pipelineruns

### CODE REVIEW

File: pkg/pipelines/tasks/set_commit_status_task.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+				createTaskParam("COMMIT_SHA", "", pipelinev1.ParamTypeString),
+				createTaskParam("DESCRIPTION", "", pipelinev1.ParamTypeString),
+				createTaskParamWithDefault("CONTEXT", "", pipelinev1.ParamTypeString, "continous-integration/tekton"),
+				createTaskParam("STATE", "", pipelinev1.ParamTypeString),
+			},
+			Steps: []v1beta1.Step{
+				{
+					Container: v1.Container{
+						Name:  "set-commit-status",
+						Image: "quay.io/cbanavik/set-commit-status:v0.1",
```
Comment (chetan-rns) [inline pkg/pipelines/tasks/set_commit_status_task.go:31]: Should move the image to `redhat-developer`

Comment (ishitasequeira) [reply pkg/pipelines/tasks/set_commit_status_task.go:31]: I guess we should move it to` redhat-developer`

**Review by wtam2018 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**